include/uncertainty_planning_core/uncertainty_contact_planning.hpp

```diff
@@ -1549,48 +1677,43 @@ class UncertaintyPlanningSpace
       const bool enable_cumulative_learning, const ConfigVector& start_configs,
       const std::function<bool(const Configuration&)>& user_goal_check_fn,
       const UncertaintyPlanningPolicyActionExecutionFunction& move_fn,
-      const double exec_time_limit, const DisplayFunction& display_fn,
-      const double policy_marker_size, const bool wait_for_user,
-      const double draw_wait) const
+      const ExecutionTimeLimit& exec_time_limit,
+      const DisplayFunction& display_fn, const double policy_marker_size,
+      const bool wait_for_user, const double draw_wait) const
   {
     const uint32_t num_executions = static_cast<uint32_t>(start_configs.size());
     UncertaintyPlanningPolicy policy = immutable_policy;
+#if UNCERTAINTY_PLANNING_CORE__SUPPORTED_ROS_VERSION == 1
     // Buffer for a teensy bit of time
     for (size_t iter = 0; iter < 100; iter++)
     {
       ros::spinOnce();
       ros::Duration(0.005).sleep();
     }
+#endif
     std::vector<ConfigVector> policy_trajectories(num_executions);
     std::vector<PolicyExecutionPerformance> policy_execution_performance(
         num_executions);
     uint32_t reached_goal = 0;
     for (uint32_t idx = 0; idx < num_executions; idx++)
     {
       Log("Starting policy execution " + std::to_string(idx) + "...", 1);
-      const double start_time = ros::Time::now().toSec();
+      ExecutionTimer exec_timer{exec_time_limit};
```

Comment:
Visually, I'd prefer to keep `Type var{value}` for initialization and `Type var(value)` for calling a constructor

Reply:
See https://github.com/calderpg/uncertainty_planning_core/pull/3/commits/cf8ce915f230816208852918241ed1d6227c0d35.